src/Bicep.Types/Concrete/ScopeType.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+
+namespace Azure.Bicep.Types.Concrete
+{
+    [Flags]
+    public enum ScopeType
+    {
+        Unknown = 0,
+
+        Tenant = 1 << 0,
+
+        ManagementGroup = 1 << 1,
+
+        Subscription = 1 << 2,
+
+        ResourceGroup = 1 << 3,
+
+        Extension = 1 << 4,
```

Comment:
Probably not super important, but do we need to model extensions of extensions as well?

Reply:
Extensions of extensions should just show up as 'ScopeType.Extension' - I'm not differentiating.